users/tests.py

```diff
@@ -25,6 +25,14 @@ def test_create_team_without_title(self):
         with pytest.raises(Exception):
             Team.objects.create(description="Best team ever")
 
+    def test_create_teams_with_same_name(self):
+        assert len(Team.objects.all()) == 0
+        Team.objects.create(name="Team1", description="This team will be created")
+        assert len(Team.objects.all()) == 1
+        with pytest.raises(Exception):
+            Team.objects.create(name="Team1", description="This team will be created")
+            assert len(Team.objects.all()) == 1, "Team1 was created twice"
```

Comment:
I had the same thought here, but because of the exception that is being raised, I can't do any operations related to the database (if I remember correctly it had something with atomic operations) outside of the "with" clause.

Reply:
you can try to run it locally and just change the line
```suggestion
            assert len(Team.objects.all()) == 10, "Team1 was created twice"
```
and you will see that the test still succeeded

[vagrant@ibm-p8-kvm-03-guest-02 vagrant]$ pytest users/tests.py -k test_create_teams_with_same_name  -v
============================================================================================================ test session starts ============================================================================================================
platform linux -- Python 3.8.2, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
django: settings: falcon.settings (from ini)
rootdir: /vagrant, configfile: pytest.ini
plugins: django-4.5.1
collected 9 items / 8 deselected / 1 selected                                                                                                                                                                                               

users/tests.py::TestTeams::test_create_teams_with_same_name PASSED                                                                                                                                                                    [100%]

means that we don't reach this line(# 34)

I also see the issue with atomic block error and it can be resolved as you can see in:
https://stackoverflow.com/a/54563945

Reply:
Worked 😄